packages/shared-ux/chrome/navigation/src/ui/components/navigation_group.tsx

```diff
@@ -45,82 +112,68 @@ function NavigationGroupInternalComp<
   Id extends string = string,
   ChildrenId extends string = Id
 >(props: Props<LinkId, Id, ChildrenId>) {
-  const { cloudLinks } = useNavigationServices();
-  const navigationContext = useNavigation();
-
-  const { children, node } = useMemo(() => {
-    const { children: _children, defaultIsCollapsed, ...rest } = props;
-    const nodeEnhanced: Omit<NodePropsEnhanced<LinkId, Id, ChildrenId>, 'children'> = {
-      ...rest,
-      isActive: defaultIsCollapsed !== undefined ? defaultIsCollapsed === false : undefined,
-      isGroup: true,
-    };
-    return {
-      children: _children,
-      node: nodeEnhanced,
-    };
-  }, [props]);
+  const { cloudLinks, navLinks$ } = useNavigationServices();
+  const { register } = useNavigation();
+  const deepLinks = useObservable(navLinks$, []);
+  const { rootIndex = 0 } = props;
 
-  const { navNode, registerChildNode, path, childrenNodes } = useInitNavNode(node, { cloudLinks });
+  const navNodeRef = useRef<ChromeProjectNavigationNode>();
+  const childrenNodesRef = useRef<ChromeProjectNavigationNode[]>();
 
-  // We add to the nav node the children that have mounted and registered themselves.
-  // Those children render in the UI inside the NavigationSectionUI -> EuiCollapsibleNavItem -> items
-  const navNodeWithChildren = useMemo(() => {
-    if (!navNode) return null;
+  const navNode = useMemo<ChromeProjectNavigationNode | null>(() => {
+    const _navNode = initNavNode(props, { cloudLinks, deepLinks });
 
-    const hasChildren = Object.keys(childrenNodes).length > 0;
-    const withChildren: ChromeProjectNavigationNode = {
-      ...navNode,
-      children: hasChildren ? Object.values(childrenNodes) : undefined,
-    };
-
-    return withChildren;
-  }, [navNode, childrenNodes]);
+    if (!_navNode) return null;
 
-  const unstyled = props.unstyled ?? navigationContext.unstyled;
+    const childrenNodes = jsxChildrenToNavigationNode(
+      { parentNodePath: _navNode.path, jsxChildren: props.children, rootIndex, treeDepth: 1 },
+      { cloudLinks, deepLinks }
+    );
 
-  const renderContent = useCallback(() => {
-    if (!path || !navNodeWithChildren) {
-      return null;
+    const childrenChanged = deepEqual(childrenNodes, childrenNodesRef.current) === false;
```

Comment:
The core deepLinks array is unstable and creates constantly new object and array references for consumers. Here I create a stable reference of the `navNode` doing a deep compare of the object.